src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -348,70 +351,14 @@ protected override void Dispose(bool disposing)
 
         private async Task<Dictionary<string, object>> CreateScriptExecutionContextAsync(object input, DataType dataType, TraceWriter traceWriter, FunctionInvocationContext invocationContext)
         {
-            // create a TraceWriter wrapper that can be exposed to Node.js
-            var log = (ScriptFunc)(p =>
-            {
-                var logData = (IDictionary<string, object>)p;
-                string message = (string)logData["msg"];
-                if (message != null)
-                {
-                    try
-                    {
-                        TraceLevel level = (TraceLevel)logData["lvl"];
-                        var evt = new TraceEvent(level, message);
-
-                        // Node captures the AsyncLocal value of the first invocation, which means that logs
-                        // are correlated incorrectly. Here we'll overwrite that value with the correct value
-                        // immediately before logging.
-                        using (BeginFunctionScope(invocationContext))
-                        {
-                            // TraceWriter already logs to ILogger
-                            traceWriter.Trace(evt);
-                        }
-                    }
-                    catch (ObjectDisposedException)
-                    {
-                        // if a function attempts to write to a disposed
-                        // TraceWriter. Might happen if a function tries to
-                        // log after calling done()
-                    }
-                }
-
-                return Task.FromResult<object>(null);
-            });
-
-            var metric = (ScriptFunc)(p =>
-            {
-                var metricObject = (IDictionary<string, object>)p;
-                string name = (string)metricObject["name"];
-                object valueObject = metricObject["value"];
-
-                if (name != null && valueObject != null)
-                {
-                    double value = Convert.ToDouble(valueObject);
-
-                    // properties are optional
-                    var properties = (IDictionary<string, object>)metricObject["properties"];
-
-                    using (BeginFunctionScope(invocationContext))
-                    {
-                        invocationContext.Logger.LogMetric(name, value, properties);
-                    }
-                }
-
-                return Task.FromResult<object>(null);
-            });
+            var traceWriterReference = new WeakReference<TraceWriter>(traceWriter);
+            var invocationContextReference = new WeakReference<FunctionInvocationContext>(invocationContext);
 
-            var bindings = new Dictionary<string, object>();
-            var bind = (ScriptFunc)(p =>
```

Comment:
This bind function goes away. In working on this fix, I was trying to minimize the bi-directional calls between .NET and Node (which is the source of the memory leaks). In this case, rather than having Node call back into .NET to process binding results, I just return them as part of a structured function result, and allow them to be processed after control returns back to .NET.